src/qpmodel.jl

```diff
@@ -152,20 +140,36 @@ function QuadraticModel(
   lvar::S = fill!(S(undef, length(c)), T(-Inf)),
   uvar::S = fill!(S(undef, length(c)), T(Inf)),
   c0::T = zero(T),
+  coo_matrices = true,
   kwargs...,
 ) where {T, S}
   ncon, nvar = size(A)
   if typeof(H) <: AbstractLinearOperator # convert A to a LinOp if A is a Matrix?
     nnzh = 0
     nnzj = 0
     data = QPData(c0, c, H, A)
-  elseif issparse(H)
-    data, nnzh, nnzj = get_QPDataCOO(c0, c, H, A)
   else
-    nnzh = typeof(H) <: DenseMatrix ? nvar * (nvar + 1) / 2 : nnz(H)
-    nnzj = nnz(A)
-    data = QPData(c0, c, H, A)
+    if coo_matrices
+      if typeof(H) <: Symmetric && !(typeof(H.data) <: SparseMatrixCOO)
+        tril!(H.data)
+        HCOO = SparseMatrixCOO(H.data)
+      elseif !(typeof(H) <: SparseMatrixCOO)
+        tril!(H)
+        HCOO = SparseMatrixCOO(H)
+      else
+        HCOO = H
+      end
+      ACOO = !(typeof(A) <: SparseMatrixCOO) ? SparseMatrixCOO(A) : ACOO = A
+      nnzh = nnz(HCOO)
+      nnzj = nnz(ACOO)
+      data = QPData(c0, c, HCOO, ACOO)
+    else
+      nnzh = typeof(H) <: DenseMatrix ? nvar * (nvar + 1) / 2 : nnz(H)
+      nnzj = nnz(A)
+      data = QPData(c0, c, H, A)
+    end
```

Comment:
What is the objective of this? If, e.g., RipQP only accepts a `QPData{T, S, SparseMatrixCOO, SparseMatrixCOO}`, then the `convert` methods should be enough, no?

Reply:
When using a Krylov method with RipQP `H` and `A` can be LinearOperators or dense Arrays. The goal is that, according to the type of QPData RipQP could solve with different algorithms (for example it cannot use a factorization with LinearOperators)